pkg/ocm/clusters.go

```diff
@@ -94,8 +94,18 @@ type OperatorIAMRole struct {
 	RoleARN   string
 }
 
-func (c *Client) HasClusters(creatorARN string) (bool, error) {
-	query := fmt.Sprintf("properties.%s = '%s'", properties.CreatorARN, creatorARN)
+// Generate a query that filters clusters running on the current AWS session account
+func getClusterFilter(creator *aws.Creator) string {
+	return fmt.Sprintf(
+		"product.id = 'rosa' AND (properties.%s LIKE '%%:%s:%%' OR aws.sts.role_arn LIKE '%%:%s:%%')",
+		properties.CreatorARN,
+		creator.AccountID,
+		creator.AccountID,
+	)
+}
```

Comment:
@jharrington22 @acathrow - this is the actual change in filtering here. It returns clusters within the same OCM org that are installed in the same AWS account ID.